### PR TITLE
Add preview for default data

### PR DESCRIPTION
**Pull request by vegaro**
A bit hacky but this way we can see the preview of the default data

image

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt [hunk 1]
```diff
@@ -130,17 +133,25 @@ private fun AppIcon(
...
-        context.applicationInfo.loadIcon(packageManager)
-    }
 
-    Image(
-        bitmap = appIconResId.toBitmap(config = Bitmap.Config.ARGB_8888).asImageBitmap(),
-        contentDescription = null,
-        modifier = modifier,
-        contentScale = ContentScale.Crop,
-    )
+    if (LocalInspectionMode.current) {
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt:137]: Should we reuse the `isInPreviewMode` composable? We might want to extract it to another file for that.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywallView.kt [hunk 2]
```diff
@@ -60,6 +61,10 @@ private fun getPaywallViewModel(
             offering,
             listener,
             MaterialTheme.colorScheme,
+            preview = isInPreviewMode(),
         ),
     )
 }
+
+@Composable
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywallView.kt:69]: I think you can add a `@ReadOnlyComposable` annotation here for some small optimizations.

**Review by tonidero — APPROVED** [review]

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/HelperFunctions.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.revenuecat.purchases.ui.revenuecatui.helpers
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.ui.platform.LocalInspectionMode
+
+@Composable
+@ReadOnlyComposable
+internal fun isInPreviewMode() = LocalInspectionMode.current
```
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/HelperFunctions.kt:9]: Can this be a lazy constant, so we don't evaluate it multiple times?

Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/HelperFunctions.kt:9]: Hmm I'm not sure you can declare a constant and mark it as `@Composable`, which is needed to use `LocalInspectionMode.current`. In any case, this composable shouldn't be recomputed unless the local composition value changes thanks to `@ReadOnlyComposable`.

Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/HelperFunctions.kt:9]: Oooh nice gotcha

**Pull request merged**